config/src/test/scala/com/typesafe/config/impl/ConfigDocumentTest.scala

```diff
@@ -193,6 +225,28 @@ class ConfigDocumentTest extends TestUtils {
                 assertTrue(e.getMessage.contains("ConfigDocument had an array at the root level"))
         }
         assertTrue(exceptionThrown)
+
+        exceptionThrown = false;
+        try {
+            document.hasValue("a")
+        } catch {
+            case e: Exception =>
+                exceptionThrown = true
+                assertTrue(e.isInstanceOf[ConfigException])
+                assertTrue(e.getMessage.contains("ConfigDocument had an array at the root level"))
+        }
+        assertTrue(exceptionThrown)
```

Comment:
you can use `val e = intercept[ConfigException] { document.hasValue("a") }; assertTrue(e.getMessage.contains)` here, which will check the same things with less code. (intercept[E] throws if the exception is not intercepted or doesn't have the expected type)